core/htmlparser/element.js

```diff
@@ -470,6 +470,42 @@ CKEDITOR.htmlParser.cssStyle = function() {
 			return ret;
 		},
 
+		/**
+		 * Searches through the current element children to find the first child matching the `criteria`.
+		 *
+		 * ```javascript
+		 * element.findOne( function( child ) {
+		 *     return child.name === 'span' || child.name === 'strong';
+		 * } ); // Will return first child which is a span or a strong.
+		 * ```
+		 *
+		 * @param {String/Function} criteria Tag name or evaluator function.
+		 * @param {CKEDITOR.htmlParser.node} criteria.child The currently iterated child.
+		 * @param {Boolean} [recursive=false] If set to `true` will iterate all descendants, otherwise only direct children.
+		 * @returns {CKEDITOR.htmlParser.node} First matched child, `null` otherwise.
+		 */
+		findOne: function( criteria, recursive ) {
+			var findMethod = typeof criteria === 'function' ? criteria : function( item ) {
```

Comment:
By inlining I meant "inline it directly into `CKEDITOR.tools.array.find` callback":

```javascript
var isMatching = typeof criteria === 'function' ? criteria( child ) : item.name === criteria;
```

IMO such version is still readable and dedicated function `findMethod` is not necessary in this case.